mungegithub/mungers/submit-queue.go

```diff
@@ -1140,6 +1140,13 @@ func (sq *SubmitQueue) doGithubE2EAndMerge(obj *github.MungeObject) bool {
 		return true
 	}
 
+	if len(sq.RequiredRetestContexts) == 0 && len(sq.BlockingJobNames) == 0 && len(sq.WeakStableJobNames) == 0 {
```

Comment:
I'd kinda rather we moved all of the 'retest' stuff around https://github.com/kubernetes/contrib/pull/1474/files#diff-2ab7cb9ccaa72dce30323c5555c00c66L1152 into its own function which just returned immediately if there was nothing to retest. And make sure that `sq.E2EStalbe()` returns quickly.